lucene/core/src/java/org/apache/lucene/codecs/blocktree/BlockTreeTermsWriter.java

```diff
@@ -709,7 +709,7 @@ private PendingBlock writeBlock(int prefixLength, boolean isFloor, int floorLead
 
           PendingTerm term = (PendingTerm) ent;
 
-          assert StringHelper.startsWith(term.termBytes, prefix): "term.term=" + term.termBytes + " prefix=" + prefix;
+          assert StringHelper.startsWith(term.termBytes, prefix): "term.term=" + new String(term.termBytes) + " prefix=" + prefix;
```

Comment:
This is wrong, uses default locale.

Reply:
Are these UTF-8? I wasn't sure, and hoped somebody would let me know during review.

Reply:
They should be. You can also dump it as a byte array for consistency with other changes you made.